src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php

```diff
@@ -176,19 +179,28 @@ protected function getContainerBuilder()
             return $this->containerBuilder;
         }
 
-        if (!$this->getApplication()->getKernel()->isDebug()) {
+        $kernel = $this->getApplication()->getKernel();
+
+        if (!$kernel->isDebug()) {
             throw new \LogicException('Debug information about the container is only available in debug mode.');
         }
 
-        if (!is_file($cachedFile = $this->getContainer()->getParameter('debug.container.dump'))) {
-            throw new \LogicException('Debug information about the container could not be found. Please clear the cache and try again.');
+        if (is_file($cachedFile = $this->getContainer()->getParameter('debug.container.dump'))) {
+            (new XmlFileLoader($container = new ContainerBuilder(), new FileLocator()))->load($cachedFile);
+        } else {
+            $buildContainer = \Closure::bind(function () { return $this->buildContainer(); }, $kernel, get_class($kernel));
+            $container = $buildContainer();
+            $container->getCompilerPassConfig()->setRemovingPasses(array());
+            $container->compile();
+            $filesystem = new Filesystem();
+            try {
+                $filesystem->dumpFile($cachedFile, (new XmlDumper($container))->dump());
```

Comment:
The goal is indeed to make the next call faster, as it is currently. About invalidation, it seems to work naturally when clearing the cache, maybe due to the fact this dump is named `%kernel.cache_dir%/%kernel.container_class%.xml`? Not sure if it's worth dumping it at all, I think it's relevant to don't need to clear the cache for getting debug informations. WDYT?